### DISCUSSION TITLE
gnpy-transmission-example /  sim_params.json

### DISCUSSION THREAD

**Discussion by devigili** [Telecominfraproject/oopt-gnpy#400] [Q&A] [answered]
Dear contributors,
I am a graduate student and I am using your QoT-E to perform some simulations in my thesis.
I am gaining confidence with the tool thanks to the youtube video uploaded online. I performed my first simulations using the branch 2020-ofc because it contains the `examples/transmission_main_example.py`.  I would like to use the newest version 2.2
however in the master branch I can not find  the `gnpy-transmission-example ` and neither the npy-path-request.

While simulating with the branch ofc-2020 I am not able to obtain the results with both the gn and the ggn model in order to compare them. Eventhough I am passing the  `--sim-params`, I obtain the same simulation results  setting `"flag_raman": false,` or `"flag_raman": true,`. and as well setting `"nli_method_name": "gn_model_analytic", `or `"nli_method_name": "gn_model_analytic",`. Probably I am not understanding properly these parameters. Since I have 35 channels  i have set `"raman_computed_channels": [1,35],`  Does `"wdm_grid_size" indicate the superchannel bandwidth?
`

Finally I have even tried to change manually `Fiber `to `RamanFiber` in the topology .json file but doing so I got one error (see the attached Fig).  Could be possible to disable the auto-design feature?

Thank you very much for your patience.

![error_RamanFiber_public](https://user-images.githubusercontent.com/84540669/119698573-a84a4480-be51-11eb-8b66-7ff58f15e070.png)

**Comment by ojnas** [comment, marked as answer]
Hi @GanbaruDM,

`gnpy-transmission-example` and `gnpy-path-request` will be available if you install the Python package according to instructions here:

https://gnpy.readthedocs.io/en/master/install.html#using-python-on-your-computer

Specifically, this step is required:

`pip install --editable . # note the trailing dot`

Note that the `gnpy-transmission-example` and `gnpy-path-request` scripts just import the corresponding function from `gnpy.tools.cli_examples` and run it.

Currently, Raman simulation and ggn model are only supported by the `RamanFiber` element. I don't see any attached file so I don't know what error you get.

**Reply by devigili** [reply]
Thanks for the prompt response.

**Reply by ojnas** [reply]
Thanks, looks like you discovered a bug in the Raman code. If a RamanFiber is shorter than the space_resolution defined in the sim-params file, the simulation crashes. Could you confirm if this is the case in your simulation? Anyway, a proposed fix for this is waiting here:

https://review.gerrithub.io/c/Telecominfraproject/oopt-gnpy/+/517505

**Reply by devigili** [reply]
Yes, I confirm that this is the case! Setting the resolution as `"space_resolution": 10e3,` (10 km, which is the length of the shorter RamanFiber span) the QoT-E does not get any error and I obtain the GGN model values.

**Reply by devigili** [reply]
I tried to implement the fix following your solution: I changed  z_resolution = self.raman_params.space_resolution to z_resolution = min(self.raman_params.space_resolution, fiber_length) in gnpy/core/science_utils.py however for some reason this is not solving the issue. Anyway I can easily set the space resolution manually to avoid any problems.
Thanks

**Reply by ojnas** [reply]
> I tried to implement the fix following your solution: I changed z_resolution = self.raman_params.space_resolution to z_resolution = min(self.raman_params.space_resolution, fiber_length) in gnpy/core/science_utils.py however for some reason this is not solving the issue.

Did you pip install GNPy from a cloned local directory with the `--editable` option? Without this option, changes you make in your local directory will not have any effect.

**Reply by jktjkt** [reply]
Just FYI, the fix has been merged. If you're using a git checkout and have installed locally before via `pip install -e .`, a simple `git pull`will do.
